sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/StateSpec.java

```diff
@@ -36,4 +37,10 @@
    * Use the {@code binder} to create an instance of {@code StateT} appropriate for this address.
    */
   StateT bind(String id, StateBinder<? extends K> binder);
+
+  /**
+   * Given {code coders} are inferred from type arguments defined for this class.
+   * @param coders Array of coders indexed by the type arguments order.
+   */
+  void offerCoders(Coder[] coders);
```

Comment:
I agree that would be better. The thing is that the `StateSpec` implementations do not hold a reference to the type variable.